packages/baseai/src/dev/routes/v1/pipes/run.ts

```diff
@@ -0,0 +1,135 @@
+import { ApiError, ApiErrorZod } from '@/dev/hono/errors';
+import { callLLM } from '@/dev/llms/call-llm';
+import { dlog } from '@/dev/utils/dlog';
+import { handleStreamingResponse } from '@/dev/utils/provider-handlers/streaming-response-handler';
+import { logger } from '@/utils/logger-utils';
+import { Hono } from 'hono';
+import { schemaMessage, VariablesSchema } from 'types/pipe';
+import { z } from 'zod';
+
+// Schema definitions
+const PipeSchema = z.object({
+	name: z.string(),
+	description: z.string().optional(),
+	status: z.string(),
+	model: z.string().default('openai:gpt-4.0-mini'),
+	stream: z.boolean().optional(),
+	json: z.boolean().optional(),
+	store: z.boolean().optional(),
+	moderate: z.boolean().optional(),
+	top_p: z.number().optional(),
+	max_tokens: z.number().optional(),
+	temperature: z.number().optional(),
+	presence_penalty: z.number().optional(),
+	frequency_penalty: z.number().optional(),
+	stop: z.array(z.string()).optional(),
+	tool_choice: z.string(),
+	parallel_tool_calls: z.boolean(),
+	messages: z.array(schemaMessage),
+	tools: z.array(z.unknown()).default([]),
+	memory: z.array(z.object({ name: z.string().trim().min(1) })).default([]),
+	variables: VariablesSchema
+});
+
+export type Pipe = z.infer<typeof PipeSchema>;
```

Comment:
We have a correct Pipe type present inside `packages/baseai/types/pipes.ts`. Why not use from there instead?